public/components/health-check/health-check.tsx

```diff
@@ -13,399 +13,407 @@ import { WAZUH_MONITORING_PATTERN } from '../../../common/constants';
-                return error;
+            const i = this.state.results.map(item => item.id).indexOf(1);
+            if (api.groups.version !== appSplit[0] || api.groups.minor !== appSplit[1]) {
+              errors.push(
+                'API version mismatch. Expected v' +
+                setupData.data.data['app-version']
+              );
+              results[i].description = <span><EuiIcon type="alert" color="danger" ></EuiIcon> Error</span>;
+              this.setState({ results, errors });
             } else {
-                this.handleError(error);
+              results[i].description = <span><EuiIcon type="check" color="secondary" ></EuiIcon> Ready</span>;
+              this.setState({ results, errors });
             }
+          }
         }
+      }
+      return;
+    } catch (error) {
+      results[0].description = <span><EuiIcon type="alert" color="danger" ></EuiIcon> Error</span>;
+      results[1].description = <span><EuiIcon type="alert" color="danger" ></EuiIcon> Error</span>;
+      this.setState({ results });
+      AppState.removeNavigation();
+      if (error && error.data && error.data.code && error.data.code === 3002) {
+        return error;
+      } else {
+        this.handleError(error);
+      }
     }
+  }
 
-    /**
-     * This check if the pattern exist then create if not
-     * @param pattern string
-     */
-    async checkSupportPattern(pattern, itemId) {
-        let results = this.state.results;
-        let errors = this.state.errors;
-        const result = await SavedObject.existsIndexPattern(pattern);
-        if (!result.data) {
-            const toast = getToasts().addWarning(`${pattern} index pattern was not found and it will be created`)
-            const fields = await SavedObject.getIndicesFields(pattern);
-            try {
-                await SavedObject.createSavedObject(
-                    'index-pattern',
-                    pattern,
-                    {
-                    attributes: {
-                        title: pattern,
-                        timeFieldName: 'timestamp'
-                    }
-                    },
-                    fields
-                );
-                getToasts().remove(toast.id);
-                getToasts().addSuccess(`${pattern} index pattern created successfully`)
-                results[itemId].description = <span><EuiIcon type="check" color="secondary" ></EuiIcon> Ready</span>;
-                this.setState({ results });
-            } catch (error) {
-                getToasts().remove(toast.id);
-                errors.push(`Error trying to create ${pattern} index pattern: ${error.message}`);
-                results[itemId].description = <span><EuiIcon type="alert" color="danger" ></EuiIcon> Error</span>;
-                this.setState({ results });
+  /**
+   * This check if the pattern exist then create if not
+   * @param pattern string
+   */
+  async checkSupportPattern(pattern, itemId) {
```

Comment:
to avoid use a relative index (`itemId`) you can receive directly the index type as parameter
```js
async checkSupportPattern(pattern, indexType) {
...
  if (!result.data) {
      const toast = getToasts().addWarning(`${pattern} index pattern was not found and it will be created`)
      const fields = await SavedObject.getIndicesFields(pattern, indexType);
...
}
....
 this.checkSupportPattern(configuration['wazuh.monitoring.pattern'], WAZUH_INDEX_TYPE_MONITORING),
 this.checkSupportPattern(`${configuration['cron.prefix']}-${configuration['cron.statistics.index.name']}-*`, WAZUH_INDEX_TYPE_STATISTICS),
```